### ISSUE TITLE
Estimate Shipping - shipping provider warnings not displayed

### ISSUE THREAD

**Issue by holydk** [nopSolutions/nopCommerce#4943]
1. nopCommerce version 4.30
2. Expected behavior - error/warning messages from shipping providers should be displayed
3. Actual behavior - if a shipping provider plugin returns any errors, they are ignored
4. Steps to reproduce the problem
-   Select "Estimate Shipping" button on the cart page.  The dialog pops up.
-   Enter address information
-   If the shipping provider plugin reports any errors (like UPS indicating invalid address, or Shipping Director's Error/ErrorExit rule returning a warning message like "call for shipping"), they are not displayed.

(Note that if you blank out the zip/postal code field, you do see an error message for "Zip / postal code is required", so the dialog does support the display of messages)

The code does not look right.  Only when 'Success' is true does it enter the block where Warnings are set:

```\Presentation\Nop.Web\Factories\ShoppingCartModelFactory.cs```

```csharp
public virtual EstimateShippingResultModel PrepareEstimateShippingResultModel(...)
...
var getShippingOptionResponse = _shippingService.GetShippingOptions(...);
if (getShippingOptionResponse.Success)
{
    if (getShippingOptionResponse.ShippingOptions.Any())
    {
        foreach (var shippingOption in getShippingOptionResponse.ShippingOptions)
        {...}
    }
    else
        foreach (var error in getShippingOptionResponse.Errors)
            model.Warnings.Add(error);
}
```

see also: https://www.nopcommerce.com/en/boards/topic/84921/estimate-shipping-on-cart-page-for-430-shipping-provider-warnings-not-displayed

**Comment by badijm** [comment]
Hi, I can work on this item, or it is for nop internal team?

**Comment by AndreiMaz** [comment]
@badijm Please feel free to create a pull request

**Comment by holydk** [comment]
Closed #4943
